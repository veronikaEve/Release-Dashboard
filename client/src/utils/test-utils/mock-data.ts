import { VersionData } from "../../@types";

export const mockVersionData: VersionData[] = [
  {
    associatedUser: "@johnSmith",
    date: "11/03/2022",
    versionFrom: "v7.8.2",
    versionTo: "v7.9.6",
    _id: "63122a332e9904df79c834ff",
  },
  {
    associatedUser: "@billy1234",
    date: "09/05/2022",
    versionFrom: "v7.7.10",
    versionTo: "v7.8.2",
    _id: "6313b230e6c802832ffeee01",
  },
];
