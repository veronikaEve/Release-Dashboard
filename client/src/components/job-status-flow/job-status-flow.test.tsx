import { render, screen } from "@testing-library/react";
import { mockActionStatusData } from "../../utils/test-utils/mock-data";
import JobStatusFlow from "./job-status-flow";

describe("JobStatusFlow", () => {
  let props = {
    statuses: mockActionStatusData,
  };

  test("should match snapshot", () => {
    const { asFragment } = render(<JobStatusFlow {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("status should have 'success' className if statusCode is success", () => {
    render(<JobStatusFlow {...props} />);

    const successStatus = screen.getByTestId("success-status");
    expect(successStatus).toHaveClass("success");
  });

  test("status should have 'error' className if statusCode is error", () => {
    render(<JobStatusFlow {...props} />);

    const errorStatus = screen.getByTestId("error-status");
    expect(errorStatus).toHaveClass("error");
  });

  test("should not display a default message if there is no status", () => {
    render(<JobStatusFlow {...props} />);

    const defaultMessage = screen.queryByText("Waiting for action ...");
    expect(defaultMessage).toBeNull();
  });

  test("should display a default message if there is no status", () => {
    props = {
      statuses: [],
    };
    render(<JobStatusFlow {...props} />);

    const defaultMessage = screen.getByText("Waiting for action ...");
    expect(defaultMessage).toBeInTheDocument();
  });
});
