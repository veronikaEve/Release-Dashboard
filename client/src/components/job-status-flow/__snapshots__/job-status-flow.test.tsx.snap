// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobStatusFlow should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="job-status-flow-container container"
  >
    <div
      class="left-stick"
    />
    <div
      class="job-statuses container-fluid"
    >
      <div
        class="job_statuses__status success"
        data-testid="success-status"
      >
        Success test message
      </div>
      <div
        class="job_statuses__status error"
        data-testid="error-status"
      >
        Error test message
      </div>
    </div>
  </div>
</DocumentFragment>
`;
